llvm/test/ExecutionEngine/JITLink/x86-64/ELF_R_X86_64_PC8.s

```diff
@@ -10,10 +10,4 @@
 main:
 	xorl	%eax, %eax
 	retq
-	.size	main, .-main
-
-	.type	P,@object
```

Comment:
Thanks for spotting the issue and providing the fix. I'm not sure why `loope` would make `llvm-mc` emit relocation and `.byte main-.` would not. I suggest not to rely on the `loope` behavior and issue `.byte main-.` in a different section, e.g. `.rodata`. I.e. if you insert `.rodata` before `.globl P`, it will force `llvm-mc` to always emit a relocation.

Reply:
Thanks for the suggestion! I fixed it by emitting a byte `main - .` to the `.rodata` section.